entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -289,20 +290,126 @@ private void doUpdate(RequestContext requestContext, EntityUpdateRequest request
       AttributeValue attributeValue =
           EntityQueryConverter.convertToAttributeValue(setAttribute.getValue()).build();
       String jsonValue = DocStoreJsonFormat.printer().print(attributeValue);
+      JSONDocument jsonDocument = new JSONDocument(jsonValue);
 
+      Map<Key, Map<String, Document>> entitiesUpdateMap = new HashMap<>();
```

Comment:
I believe this entire logic can reuse the logic in `transformUpdateOperations`. 

Can you please see, if `transformUpdateOperations` can be re-usable both by `doUpdate` and `doBulkUpdate`? I think I did have a similar review comment before https://github.com/hypertrace/entity-service/pull/121#discussion_r682316361

Reply:
In `transformUpdateOperations` we are computing the `subDocPath ` and `jsonDocument.` 
The `Update` api only supports update operations where we update the same attribute with same value for multiple entities. So if we use `transformUpdateOperations` in `doUpdate` then we will unnecessarily compute `subDocPath` and `jsonDocument` again and again.